samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -66,6 +67,14 @@ public IncomingMessageEnvelope(SystemStreamPartition systemStreamPartition, Stri
     this.size = size;
   }
 
+  public void setTimestamp(long timestamp) {
```

Comment:
Just wondering if it will be clearer to call this as `setProcessingTimestamp()` or `setArrivalTimestamp()`. 

Timestamp sounds abstract enough here and could be interpreted in multiple ways here.

Would be great if we can add annotation `InterfaceStability.Evolving`  to the signature and java doc.